hw04-Proxy/src/main/java/ru/otus/homework/ProxySample.java

```diff
@@ -0,0 +1,38 @@
+package ru.otus.homework;
+
+import ru.otus.homework.logging.Log;
+import ru.otus.homework.logging.Proxibale;
+import ru.otus.homework.logging.TestLogging;
+import ru.otus.homework.logging.TestLoggingInterface;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.ArrayList;
+
+public class ProxySample {
+    static <T extends Proxibale> T createClass(T someInterface) throws ClassNotFoundException {
+        ArrayList<String> methodsToInvoke = new ArrayList<>();
```

Comment:
```suggestion
        List<String> methodsToInvoke = new ArrayList<>();
```
Лучше определять как интерфейсы, чтобы проще было заменить реализацию в случае необходимости.

Еще подумайте, какая структура данных эффективнее работает с `contains()`

Reply:
Очень интересное замечание, не подумал про более подходящую структуру данных. В данном случае это была бы HashSet

Reply:
Да, HashSet
здесь интересное сравнение contains в ArrayList и HashSet -- https://www.baeldung.com/java-hashset-arraylist-contains-performance